app/components/channel_drawer/channels_list/filtered_list/filtered_list.js

```diff
@@ -116,72 +119,92 @@ class FilteredList extends Component {
-        const {
-            unreadChannels,
-            favoriteChannels,
-            publicChannels,
-            privateChannels
-        } = props.channels;
+    getSectionBuilders = () => ({
+        unreads: {
+            builder: this.buildUnreadChannelsForSearch,
+            id: 'mobile.channel_list.unreads',
+            defaultMessage: 'UNREADS'
+        },
+        channels: {
+            builder: this.buildChannelsForSearch,
+            id: 'sidebar.channels',
+            defaultMessage: 'CHANNELS'
+        },
+        dms: {
+            builder: this.buildCurrentDMSForSearch,
+            id: 'sidebar.direct',
+            defaultMessage: 'DIRECT MESSAGES'
+        },
+        members: {
+            builder: this.buildMembersForSearch,
+            id: 'mobile.channel_list.members',
+            defaultMessage: 'MEMBERS'
+        },
+        nonmembers: {
+            builder: this.buildOtherMembersForSearch,
+            id: 'mobile.channel_list.not_member',
+            defaultMessage: 'NOT A MEMBER'
+        }
+    });
 
-        const notMemberOf = otherChannels.map((o) => {
-            return {
-                ...o,
-                fake: true
-            };
+    buildUnreadChannelsForSearch = (props, term) => {
+        const {unreadChannels} = props.channels;
+
+        return this.filterChannels(unreadChannels, term);
+    }
+
+    buildCurrentDMSForSearch = (props, term) => {
+        const {channels, teammateNameDisplay, profiles, statuses, pastDirectMessages} = props;
+        const {favoriteChannels} = channels;
+
+        const favoriteDms = favoriteChannels.filter((c) => {
+            return c.type === General.DM_CHANNEL;
         });
 
-        const favorites = favoriteChannels.filter((c) => {
-            return c.type !== General.DM_CHANNEL && c.type !== General.GM_CHANNEL;
+        const directChannelUsers = [];
+        const groupChannels = [];
+
+        channels.directAndGroupChannels.forEach((c) => {
+            if (c.type === General.DM_CHANNEL) {
+                directChannelUsers.push(profiles[c.teammate_id]);
+            } else {
+                groupChannels.push(c);
+            }
         });
 
-        const unreads = this.filterChannels(unreadChannels, term);
-        const channels = this.filterChannels([...favorites, ...publicChannels, ...privateChannels], term).
-            sort(sortChannelsByDisplayName.bind(null, props.intl.locale));
+        const pastDirectMessageUsers = pastDirectMessages.map((p) => profiles[p]).filter((p) => typeof p !== 'undefined');
 
-        const others = this.filterChannels(notMemberOf, term);
-        const groups = this.filterChannels(groupChannels, term);
-        const fakeDms = this.filterChannels(this.buildFakeDms(props), term);
-        const directMessages = [...groups, ...fakeDms].sort(sortChannelsByDisplayName.bind(null, props.intl.locale));
+        const dms = [...directChannelUsers, ...pastDirectMessageUsers].map((u) => {
```

Comment:
So they will not show up as a duplicate here, but they did show up in the members section of the search. Added a check to make sure this doesn't happen.